### PR TITLE
Skip adding loganalyzer marker if loganalyzer is disabled

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

The `loganlyzer` fixture still adds marker to syslog when log analyzer is disabled. This PR improves this behavior.
When log anlayzer is disabled, do nothing except output a log message indicating that log analyzer is disabled.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
When log analyzer is disabled, it's unnecessary to add start marker to syslog. This PR can save some script execution time in setup/teardown sections of test case when log analyzer is disabled. The log messages would be fewer too.

#### How did you do it?
Updated the `loganalyzer` fixture. When log analzyer is disabled, skip adding start marker to DUT's syslog.

#### How did you verify/test it?
Tested scenarios:
* Disable log analyzer from pytest cli
  * Run regular test case
  * Run test case explicitly called log analyzer
  * Run test case with `@pytest.mark.disable_loganalyzer
* Disable log analyzer from test script by adding pytest.mark.disable_loganalyzer
  * Run regular test case
  * Run test case explicitly called log analyzer
  * Run test case with `@pytest.mark.disable_loganalyzer
* Do not disable log analyzer
  * Run regular test case
  * Run test case explicitly called log analyzer
  * Run test case with `@pytest.mark.disable_loganalyzer

### CODE REVIEW

**Comment by wangxin** [comment]
@yvolynets-mlnx Could you please have a look?

**Comment by yvolynets-mlnx** [comment]
> 
> 
> @yvolynets-mlnx Could you please have a look?

Looks good

File: tests/common/plugins/loganalyzer/__init__.py [hunk 1]
```diff
@@ -11,6 +11,10 @@ def pytest_addoption(parser):
 
 @pytest.fixture(autouse=True)
 def loganalyzer(duthost, request):
+    if request.config.getoption("--disable_loganalyzer") or "disable_loganalyzer" in request.keywords:
+        logging.info("Log analyzer is disabled")
+        yield
```
Comment (tahmed-dev) [inline tests/common/plugins/loganalyzer/__init__.py:16]: no need to yield, we can just return

Comment (wangxin) [reply tests/common/plugins/loganalyzer/__init__.py]: Thanks! Updated.

**Comment by wangxin** [comment]
retest this please

**Comment by wangxin** [comment]
retest this please

**Comment by wangxin** [comment]
@tahmed-dev It turns out that the `yield` is mandatory. Otherwise it won't work. I have added back the `yield`.

**Comment by wangxin** [comment]
retest this please

**Comment by wangxin** [comment]
retest vsimage please

**Pull request merged**